Tokens/phpBBToken.php

```diff
@@ -10,12 +10,78 @@
 namespace phpBB\SessionsAuthBundle\Tokens;
 
 
-use Symfony\Component\Security\Core\Authentication\Token\PreAuthenticatedToken;
+use Symfony\Component\Security\Core\Authentication\Token\AbstractToken;
+use Symfony\Component\Security\Core\User\UserInterface;
 
-class phpBBToken extends PreAuthenticatedToken{
-    public function __construct($user, $credentials, $providerKey, array $roles = array())
+class phpBBToken extends AbstractToken
+{
+    private $providerKey;
+
+    /**
+     * Constructor.
+     * @param string|UserInterface $user
+     * @param $providerKey
+     * @param array $roles
+     */
+    public function __construct($user, $providerKey, array $roles = array())
+    {
+        parent::__construct($roles);
+
+        if (empty($providerKey))
+        {
+            throw new \InvalidArgumentException('$providerKey must not be empty.');
+        }
+
+        $this->setUser($user);
+        $this->providerKey = $providerKey;
+
+        if (!empty($roles))
+        {
+            $this->setAuthenticated(true);
```

Comment:
No role == unauthenticated in symfony.
anonymous has always a role, the anonymous role.